### ISSUE TITLE
Allow ID columns to be named other than "ID"

### ISSUE THREAD

**Issue by kbroman** [rqtl/qtl2geno#53]
The [B6xBTBR data](http://phenome.jax.org/db/q?rtn=projects/projdet&reqprojid=532) uses `"MouseNum"` as the ID. Maybe specify the column name in the control file (or just take the first column in each file)?

In preparing these data for R/qtl2, I'm getting the following error, which is not too informative:

```
> b6btbr <- qtl2geno::read_cross2("b6btbr.json")
 Error in check4duplicates(idcolumn, name) :
   Not all ids in  are unique: "NA"
```

It looks like the covariate file that I'd prepared has the following at the bottom:

```
Mouse3158,Male,1,12/19/2007,8/11/2008,NA,11/28/2007,07/15/2008,12/05/2007
Mouse3294,Female,1,12/18/2007,8/11/2008,11/26/2007,11/28/2007,NA,12/10/2007
NA,NA,NA,12/18/2007,8/11/2008,NA,11/28/2007,NA,12/05/2007
NA,NA,NA,NA,NA,NA,NA,07/15/2008,NA
NA,NA,NA,NA,NA,NA,NA,07/15/2008,NA
```

There are maybe some extra individuals in some of the microarray batch files?

**Comment by kbroman** [comment]
Actually, the code allows the first column to have any name. It totally ignores the name and just grabs it as a set of IDs.

**Comment by kbroman** [comment]
Fixed in PR #54. The problem was in the call from `read_csv()` to `firstcol2rownames()` (need to pass file name) and then in the call from `firstcol2rownames()` to `check4duplicates()` (the error message in `check4duplicates()` needed to be improved).
